ignite/services/plugin/plugin.go

```diff
@@ -155,8 +159,15 @@ func newPlugin(pluginsDir string, cp pluginsconfig.Plugin) *Plugin {
 }
 
 func (p *Plugin) KillClient() {
-	if p.client != nil {
-		p.client.Kill()
+	if p.isHost {
+		if p.client != nil {
+			p.client.Kill()
+		}
+		DeletePluginConfCache(p.Path)
+		p.isHost = false
+		// if the sharedHost config parameter is set to false we shutdown normally.
+	} else if !p.Plugin.SharedHost {
+		DeletePluginConfCache(p.Path)
```

Comment:
Why not killing the plugins that have `isHost=false` ? Since `isHost` is true only for the first *sharedHost* plugin, all the legacy plugins won't be killed right ?

In addition I don't understand why you're deleting the cache for *non-SharedHost* plugins, because they don't use it anyway.

I would write it like that :

```suggestion
	if p.isHost {
		DeletePluginConfCache(p.Path)
	}
	if p.client != nil {
		p.client.Kill()
	}
```

WDYT?

Reply:
There needs to be safe guarding against a plugin killing the host server when the plugin is shutting down. we defiantly don't need to delete from the cache is the flag isn't set however. Something like what is below I think is better: 
```go
if !p.Plugin.SharedHost {
  if p.client != nil {
       p.client.Kill()
   }
    return
  }

if p.isHost {
  if p.client != nil {
   p.client.Kill()
}
DeletePluginConfCache(p.Path)
 p.isHost = false
  // if the sharedHost config parameter is set to false we shutdown n normally.
}
```

Reply:
this is a better execution flow. I was against a complex condition in favor of two simpler ones. In regards to `p.isHost = false` I think its safer to set the flag to false if in the future there are post shutdown operations that will need to elevate another plugin instance to `host` we can remove this flag change for now until its relevant but I was thinking it was better to add it now to save debugging in the future.

Reply:
OK for me.